src/pages/pupils/beta/years/[yearSlug]/subjects.tsx

```diff
@@ -0,0 +1,136 @@
+import { GetStaticPathsResult, GetStaticProps } from "next";
+import Link from "next/link";
+
+import { resolveOakHref } from "@/common-lib/urls";
+import curriculumApi2023 from "@/node-lib/curriculum-api-2023";
+import { PupilSubjectListingData } from "@/node-lib/curriculum-api-2023/queries/pupilSubjectListing/pupilSubjectListing.schema";
+import getPageProps from "@/node-lib/getPageProps";
+import {
+  getFallbackBlockingConfig,
+  shouldSkipInitialBuild,
+} from "@/node-lib/isr";
+
+type SubjectListingPageProps = { curriculumData: PupilSubjectListingData[] };
+
+const PupilSubjectListing = (props: SubjectListingPageProps) => {
+  const { curriculumData } = props;
+  const subjectObject: Record<string, PupilSubjectListingData[]> =
```

Comment:
this would be easier to understand if it was called groupedBySubject
also worth looking at these sort of things https://lodash.com/docs/4.17.15#groupBy for not reinventing the wheel

Reply:
yes!  i was trying to find that, thanks